packages/lib/passport-interface/src/api/requestPodboxTicketActionPreCheck.ts

```diff
@@ -0,0 +1,47 @@
+import { SerializedPCD } from "@pcd/pcd-types";
+import { SemaphoreSignaturePCD } from "@pcd/semaphore-signature-pcd";
+import urlJoin from "url-join";
+import {
+  ActionConfigResponseValue,
+  PodboxTicketActionPreCheckRequest
+} from "../RequestTypes";
+import { PodboxTicketAction } from "../TicketAction";
+import { APIResult } from "./apiResult";
+import { httpPostSimple } from "./makeRequest";
+
+/**
+ * Prior to performing an action such as check-in on a ticket, retrieve a list
+ * of supported actions, if any. If the credential does not authorize the
+ * performance of actions on this ticket, the return value will indicate this.
+ *
+ * {@link credential} is a Semaphore Signature of a payload that is a
+ * `JSON.stringify`-ed {@link CredentialPayload}.
+ *
+ * Never rejects. All information encoded in the resolved response.
+ */
+export async function requestPodboxTicketActionPreCheck(
+  preCheckUrl: string,
+  credential: SerializedPCD<SemaphoreSignaturePCD>,
```

Comment:
did you want to use the `Credential` interface here? it would also let you delete your comment on lines 17-18

Reply:
Yes, I've eliminated the last few non-uses of this.